lua/auto-save/init.lua

```diff
@@ -124,7 +127,9 @@ function M.on()
   })
   api.nvim_create_autocmd(cnf.opts.trigger_events.cancel_defered_save, {
     callback = function (opts)
-      cancel_timer(opts.buf)
+      if should_be_saved(opts.buf) then
+        cancel_timer(opts.buf)
+      end
```

Comment:
Good question. Since the `cancel_timer` function is used in `defer_save` as well, we would check 2 times for each defer save. That **could** be problematic for larger user given conditions.